docs/content/posts/2025-07-31-release-2.1.0.md

```diff
@@ -0,0 +1,465 @@
+propagated join output.
+
+This "zero intermediate state" approach primarily targets state reduction, offering substantial
+benefits in resource consumption and operational stability. This feature is now available for
+pipelines with multiple INNER/LEFT joins that share at least one common join key, enable with
+`SET 'table.optimizer.multi-join.enabled' = 'true'`.
+
+**Benchmark**
+
+Here's a 10-way benchmark between the default streaming join and the multi-way join optimization, 
+the star schema join pattern as following:
+```sql
+-- Enable multi-way join optimization
+SET 'table.optimizer.multi-join.enabled' = 'true';
+
+-- star schema join pattern
+INSERT INTO JoinResultsMJ SELECT * FROM TenantKafka t
+LEFT JOIN SuppliersKafka s ON t.tenant_id = s.tenant_id AND ...
+LEFT JOIN ProductsKafka p ON t.tenant_id = p.tenant_id AND ...
+LEFT JOIN CategoriesKafka c ON t.tenant_id = c.tenant_id AND ...
+LEFT JOIN OrdersKafka o ON t.tenant_id = o.tenant_id AND ...
+LEFT JOIN CustomersKafka cust ON t.tenant_id = cust.tenant_id AND ...
+LEFT JOIN WarehousesKafka w ON t.tenant_id = w.tenant_id AND ...
+LEFT JOIN ShippingKafka sh ON t.tenant_id = sh.tenant_id AND ...
+LEFT JOIN PaymentKafka pay ON t.tenant_id = pay.tenant_id AND ...
+LEFT JOIN InventoryKafka i ON t.tenant_id = i.tenant_id AND ...;
+```
+
+You can observe the amount of intermediate state in the first section, the amount of records processed when the operators reach 100% busyness in the second section, and the checkpoints in the third.
+
+<div style="text-align: center;">
+<img src="/img/blog/2025-07-31-release-2.1.0/multi_way_join.png" style="width:70%;margin:15px">
+</div>
+
+For this 10-way join above, involving record amplification, the benchmark benefits as follows:
+
+- Performance: 2x to over 100x+ increase in processed records when both at 100% busyness.
+- State Size: 3x to over 1000x+ smaller as intermediate state grows.
+
+**More Information**
+* [FLINK-37859](https://issues.apache.org/jira/browse/FLINK-37859)
+* [Multi-way Join](https://cwiki.apache.org/confluence/display/FLINK/FLIP-516%3A+Multi-Way+Join+Operator)
+
+## Async Lookup Join Enhancements
+
+Support handling records in order based on upsert key (the unique key in the input stream deduced by
+planner) while allowing parallel processing of different keys to achieve better throughput when
+processing changelog data stream.
+
+**More Information**
+* [FLINK-37874](https://issues.apache.org/jira/browse/FLINK-37874)
+* [Async Lookup Join](https://cwiki.apache.org/confluence/display/FLINK/FLIP-519%3A++Introduce+async+lookup+key+ordered+mode)
+
+## Sink Reuse
+
+Within a single Flink job, when write multiple `INSERT INTO` statements update identical columns (
+different columns will be supported in next release) of a target table, the planner will optimize 
+the execution plan and merge the sink nodes to achieve reuse. This would be a great usability improvement 
```

Comment:
```suggestion
Within a single Flink job, when writing multiple `INSERT INTO` statements updating identical columns (
different columns will be supported in next release) of a target table, the planner will optimize 
the execution plan and merge the sink nodes to achieve reuse. This will be a great usability improvement 
```